logstash-core/lib/logstash/filter_delegator.rb

```diff
@@ -31,18 +34,26 @@ def config_name
     end
 
     def multi_filter(events)
-      @metric.increment(:events_in, events.size)
+      @metric_events.increment(:in, events.size)
 
       new_events = @filter.multi_filter(events)
 
-      @metric.increment(:events_out, new_events.size)
+      # There is no garantee in the context of filter
+      # that EVENTS_INT == EVENTS_OUT, see the aggregates and
+      # the split filter
+      @metric_events.increment(:out, new_events.size) unless new_events.nil?
+
       return new_events
     end
 
     private
     def define_flush_method
       define_singleton_method(:flush) do |options = {}|
-        @filter.flush(options)
+        # we also need to trace the number of events
+        # coming from a specific filters.
+        new_events = @filter.flush(options)
+        @metric_events.increment(:out, new_events.size) unless new_events.nil? # Logstash-filter-aggregates return nil.
```

Comment:
It can happen in any filters that does some kind of internal buffering like the multiline filter.
I can make the comment more generic if its the issues? I wanted to add context for the reviewer for why I was doing that check in this case.